Sources/SwiftBSON/BSON.swift

```diff
@@ -95,6 +95,26 @@ public enum BSON {
             self = int64.bson
             return
         }
+
+        if let uuidJSON = try json.unwrapObject(withKey: "$uuid", keyPath: keyPath) {
```

Comment:
I notice that the other types have the similar parsing logic abstracted, should we do something similar with `uuid`?

Reply:
The integers are done separately here to ensure they come first (see the comment above), but the rest of the parsing should be done via each type's conformance to `BSONValue`. This binary parsing logic can just go with the rest of the extended JSON parsing logic in in `BSONBinary.init(fromExtendedJSON:)` in `BSONBinary.swift`.